single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh

```diff
@@ -0,0 +1,323 @@
+    exit 1
+fi
+if [[ -z "$AZURE_RESOURCE_GROUP_NAME" ]]; then
+    echo "No Azure resource group [AZURE_RESOURCE_GROUP_NAME] specified."
+    exit 1
+fi
+if [[ -z "$AZURE_RESOURCE_GROUP_LOCATION" ]]; then
+    echo "No Azure resource group [AZURE_RESOURCE_GROUP_LOCATION] specified."
+    exit 1
+fi
+
+# Login to Azure and select the subscription
+if ! AZURE_USERNAME=$(az account show --query user.name); then
+    echo "No Azure account logged in, now trying to log in."
+    az login --output none
+    az account set --subscription "$AZURE_SUBSCRIPTION_ID"
+else
+    echo "Logged in as $AZURE_USERNAME, set the active subscription to \"$AZURE_SUBSCRIPTION_ID\""
+    az account set --subscription "$AZURE_SUBSCRIPTION_ID"
+fi
+
+# Check the resource group and region
+RG_EXISTS=$(az group exists --resource-group "$AZURE_RESOURCE_GROUP_NAME")
+if [[ $RG_EXISTS == "false" ]]; then
+    echo "Creating resource group $AZURE_RESOURCE_GROUP_NAME in $AZURE_RESOURCE_GROUP_LOCATION."
+    az group create --location "$AZURE_RESOURCE_GROUP_LOCATION" --resource-group "$AZURE_RESOURCE_GROUP_NAME" --output none
+else
+    echo "Resource group $AZURE_RESOURCE_GROUP_NAME exists in $AZURE_RESOURCE_GROUP_LOCATION."
+    RG_LOCATION=$(az group show --resource-group "$AZURE_RESOURCE_GROUP_NAME" --query location)
+    if [[ "$RG_LOCATION" != "\"$AZURE_RESOURCE_GROUP_LOCATION\"" ]]; then
+        echo "Resource group $AZURE_RESOURCE_GROUP_NAME is located in $RG_LOCATION, not \"$AZURE_RESOURCE_GROUP_LOCATION\""
+    fi
+fi
+
+# Validate the ARM templates (Jacob)
+
+securityGroupName="${DEPLOYMENT_PREFIX}nsg01"
+securityGroupLocation="$AZURE_RESOURCE_GROUP_LOCATION"
+
+echo "Validating parameters for Network Security Group..."
+if ! az deployment group validate \
+    --resource-group "$AZURE_RESOURCE_GROUP_NAME" \
+    --template-file ./securitygroup/securitygroup.template.json \
+    --parameters \
+        securityGroupName="$securityGroupName" \
+        securityGroupLocation="$securityGroupLocation" \
+    --output none; then
+    echo "Validation error for Network Security Group, please see the error above."
+    exit 1
+else
+    echo "Network Security Group parameters are valid."
+fi
+
+securityGroupName="${DEPLOYMENT_PREFIX}nsg01"
+vnetName="${DEPLOYMENT_PREFIX}vnet01"
+vnetLocation="$AZURE_RESOURCE_GROUP_LOCATION"
+
+echo "Validating parameters for Virtual Network..."
+if ! az deployment group validate \
+    --resource-group "$AZURE_RESOURCE_GROUP_NAME" \
+    --template-file ./vnet/vnet.template.json \
+    --parameters \
+        securityGroupName="$securityGroupName" \
+        vnetName="$vnetName" \
+        vnetLocation="$vnetLocation" \
+    --output none; then
+    echo "Validation error for Virtual Network, please see the error above."
+    exit 1
+else
+    echo "Virtual Network parameters are valid."
+fi
+
+tagValues="{}"
+
+securityGroupName="${DEPLOYMENT_PREFIX}nsg01"
+vnetName="${DEPLOYMENT_PREFIX}vnet01"
+disablePublicIp=false
```

Comment:
I think should come from your variables. Let me know, we could also make part of "secure cluster"

Reply:
Actually, I added only lines 88 and 89 so could you please check it with the person who was coding?

Reply:
nws